src/syscalls/native_syscall_handler.rs

```diff
@@ -87,9 +87,13 @@ impl<'a, 'cache, S: StateReader, C: ContractClassCache> StarkNetSyscallHandler
         gas: &mut u128,
     ) -> SyscallResult<cairo_vm::felt::Felt252> {
         tracing::debug!("Called `get_block_hash({block_number})` from Cairo Native");
-
         self.handle_syscall_request(gas, "get_block_hash")?;
 
+        let current_block_number = dbg!(self.block_context.block_info.block_number);
+        if block_number > current_block_number - 10 {
```

Comment:
This operation can panick, lets check that current_block_number is >= 10

Reply:
Or you can do this instead:
```suggestion
        if block_number > current_block_number.saturating_sub(10) {
```

Reply:
Or compare with an addition instead:
```suggestion
        if block_number + 10 > current_block_number {
```